mne/beamformer/_dics.py

```diff
@@ -1,32 +1,452 @@
+
+            ``None`` : orientations are pooled (Default)
+            'normal' : filters are computed for the orientation tangential to
+                       the cortical surface
+            'max-power' : filters are computer for the orientation that
+                          maximizes spectral power.
+
+    real_filter : bool
+        If ``True``, take only the real part of the cross-spectral-density
+        matrices to compute real filters. Defaults to ``False``.
+    verbose : bool, str, int, or None
+        If not ``None``, override default verbose level (see
+        :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>` for
+        more).
+
+    Returns
+    -------
+    filters : dict
+        Dictionary containing filter weights from DICS beamformer.
+        Contains the following keys:
+
+            'weights' : {array}
+                For each frequency, the filter weights of the beamformer.
+            'csd' : {instance of CrossSpectralDensity}
+                The data cross-spectral density matrices used to compute the
+                beamformer.
+            'ch_names' : {list}
+                Channels used to compute the beamformer.
+            'proj' : {array}
+                Projections used to compute the beamformer.
+            'vertices' : {list}
+                Vertices for which the filter weights were computed.
+            'n_orient' : int
+                Number of source orientations defined in the forward model.
+            'subject' : {string}
+                The subject ID.
+
+    See Also
+    --------
+    apply_dics, dics, dics_epochs
+
+    Notes
+    -----
+    For more information about ``real_filter``, see the
+    `supplemental information <http://www.cell.com/cms/attachment/616681/4982593/mmc1.pdf>`_
+    from [2]_.
+
+    References
+    ----------
+    .. [1] Gross et al. Dynamic imaging of coherent sources: Studying neural
+           interactions in the human brain. PNAS (2001) vol. 98 (2) pp. 694-699
+    .. [2] Hipp JF, Engel AK, Siegel M (2011) Oscillatory Synchronization
+           in Large-Scale Cortical Networks Predicts Perception.
+           Neuron 69:387-396.
+    """  # noqa: E501
+    allowed_ori = [None, 'normal', 'max-power']
+    if pick_ori not in allowed_ori:
+        raise ValueError('"pick_ori" should be one of %s.' % allowed_ori)
+
+    frequencies = [np.mean(freq_bin) for freq_bin in csd.frequencies]
+    n_freqs = len(frequencies)
+    n_orient = forward['sol']['ncol'] // forward['nsource']
+
+    picks = _setup_picks(info=info, forward=forward)
+    _, ch_names, proj, vertices, G = _prepare_beamformer_input(
+        info, forward, label, picks=picks, pick_ori=pick_ori
+    )
+    csd_picks = [csd.names.index(ch) for ch in ch_names]
+
+    n_sources = G.shape[1] // n_orient
+
+    # Normalize the leadfield for depth weighting. This also ensures that
+    # G @ G.T == I, which simplifies some of the equations.
+    G = G / np.linalg.norm(G, axis=0)
```

Comment:
I think the leadfield normalization should be optional (especially since there is also the possibility of weight normalization, right?). 
Also note that e.g. FieldTrip has two different ways of normalizing: https://github.com/fieldtrip/fieldtrip/blob/master/forward/ft_compute_leadfield.m#L569

Regarding https://github.com/mne-tools/mne-python/pull/4659 it might make sense to make the normalization code then available to both beamformer types.

Reply:
Interesting. Can you give me a concrete example when you would want to turn off the leadfield normalization?

Reply:
Well, first, there are different ways of doing leadfield normalization (cf. [FieldTrip](https://github.com/fieldtrip/fieldtrip/blob/master/forward/ft_compute_leadfield.m#L569) and Sekihara & Nagarajan, 2008, p. 39) - probably those should be different options.
At least with time series (LCMV), leadfield normalization can perform worse than weight normalization and thus weight normalization might be preferred over leadfield normalization.
Finally, in the case where conditions are compared, the leadfield normalization cancels out and is thus an unnecessary step.